From ebf185f8e48b5aec622dc949cef1b19b0a7669ef Mon Sep 17 00:00:00 2001
From: Giulio Benetti <giulio.benetti@benettiengineering.com>
Date: Wed, 1 Jan 2020 12:54:45 +0100
Subject: [PATCH] Bug 1606119 - Fix PPC HW Crypto build failure

All Altivec *_be() functions are supported from gcc version 8.x not 5.x
so modify gcc version check that at the moment cause build failure due
to missing Altivec *_be() functions.

Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
Upstream issue: https://bugzilla.mozilla.org/show_bug.cgi?id=1606119
---
 nss/lib/freebl/gcm.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/nss/lib/freebl/gcm.h b/nss/lib/freebl/gcm.h
index 571b9ec55..aa4dee824 100644
--- a/nss/lib/freebl/gcm.h
+++ b/nss/lib/freebl/gcm.h
@@ -41,11 +41,11 @@
 #endif
 
 /*
- * PPC CRYPTO requires at least gcc 5 or clang. The LE check is purely
+ * PPC CRYPTO requires at least gcc 8 or clang. The LE check is purely
  * because it's only been tested on LE. If you're interested in BE,
  * please send a patch.
  */
-#if (defined(__clang__) || (defined(__GNUC__) && __GNUC__ >= 5)) && \
+#if (defined(__clang__) || (defined(__GNUC__) && __GNUC__ >= 8)) && \
     defined(IS_LITTLE_ENDIAN)
 #define USE_PPC_CRYPTO
 #endif
-- 
2.20.1

